xwiki-platform-core/xwiki-platform-display/xwiki-platform-display-api/src/main/java/org/xwiki/display/internal/AbstractDocumentTitleDisplayer.java

```diff
@@ -136,9 +141,15 @@ public XDOM display(DocumentModelBridge document, DocumentDisplayerParameters pa
     private XDOM displayTitle(DocumentModelBridge document, DocumentDisplayerParameters parameters)
     {
         // 1. Try to use the title provided by the user.
-        if (!StringUtils.isEmpty(document.getTitle())) {
+        String rawTitle = document.getTitle();
+        if (!StringUtils.isEmpty(rawTitle)) {
             try {
-                return parseTitle(evaluateTitle(document.getTitle(), document.getDocumentReference(), parameters));
+                String title = rawTitle;
+                // Evaluate the title only if the document has script rights, otherwise use the raw title.
+                if (authorizationManager.hasAccess(Right.SCRIPT)) {
```

Comment:
I wonder whether it's good to check rights in the displayer component. I know it's more convenient, but it feels like we're mixing the concerns. Ideally, the component (e.g. a script service) that calls the displayer should check the rights (view, script, etc.). The displayer could have a parameter to control whether the displayed text must be evaluated or not. It's a bit more cumbersome but at lest we separate the concerns.